starky2/src/cross_table_lookup.rs

```diff
@@ -0,0 +1,294 @@
+use anyhow::{ensure, Result};
```

Comment:
In terms of naming do you think "cross-table permutation" would be okay? Lookup might be misinterpreted as a subset type lookup

Reply:
Hmm I think permutation might be confusing since the tables don't have the same dimensions.
Technically, it's a multiset subset check `looked ⊂ looking` + a check ` looking \ looked = default^(n-m)`. So maybe "cross-table multiset check` ? 
Also, right now only one-to-one lookups are supported, but in the future it might be useful to support many-to-one lookups. For example, having many tables use a single bignum table for their bignum arithmetic. In that context, I think "lookup" makes more sense.

Reply:
Hm ok seems reasonable. I guess lookup doesn't have to mean that the table contains no duplicates. Let's keep lookup/CTL then.